src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -243,84 +192,81 @@ public void buildSettlement(UserBean user, String gameIdString, String nodeIdStr
 
         if (!game.getCurrentMove().equals(gameUserBean.getMoveOrder())) {
-            log.debug("It is not current turn of user {}", user.getUsername());
+            log.debug("It is not current turn of user {}", gameUserBean.getUser().getUsername());
             throw new PlayException(ERROR_CODE_ERROR);
         }
+    }
 
-        switch (game.getStage()) {
-            case PREPARATION:
-                Integer preparationCycle = game.getPreparationCycle();
-                if (isFirstMove(game) && !isOddCycle(preparationCycle) || isLastMove(game) && isOddCycle(preparationCycle)) {
-                    if (preparationCycle == initialBuildingsSet.size()) {
-                        game.setStage(GameStage.MAIN);
-                        game.setPreparationCycle(null);
-                        log.debug("Game Stage was changed from PREPARATION to {}", game.getStage());
-                        nextMoveNumber = 1;
-                    } else {
-                        nextMoveNumber = endTurnUtil.getNextMoveInPreparationStage(game);
-                        game.setPreparationCycle(preparationCycle + 1);
-                        log.debug("Preparation cycle increased by 1. Current preparation cycle is {} of {}", game.getPreparationCycle(), initialBuildingsSet.size());
-                    }
-                } else {
-                    nextMoveNumber = endTurnUtil.getNextMoveInPreparationStage(game);
-                }
-                break;
-            case MAIN:
-                nextMoveNumber = endTurnUtil.getNextMoveInMainStage(game);
-                break;
-            default:
-                log.debug("Cannot recognize current game stage: {}", game.getStage());
-                throw new PlayException(ERROR_CODE_ERROR);
+    private void validateUserNotEmpty(UserBean user) throws PlayException {
+        if (user == null) {
+            log.debug("User should not be empty");
+            throw new PlayException(ERROR_CODE_ERROR);
         }
-
-        log.debug("Next move order in {} stage is changing from {} to {}", game.getStage(), game.getCurrentMove(), nextMoveNumber);
-        game.setCurrentMove(nextMoveNumber);
-
-        gameDao.updateGame(game);
     }
 
-    private boolean isOddCycle(Integer preparationCycle) {
-        return preparationCycle % 2 > 0;
+    private GameUserBean getJoinedGameUser(UserBean user, GameBean game) throws PlayException {
+        for (GameUserBean gameUser : game.getGameUsers()) {
+            if (gameUser.getUser().equals(user)) {
+                return gameUser;
+            }
+        }
+
+        log.debug("User is not joined to game {}", game.getGameId());
+        throw new PlayException(ERROR_CODE_ERROR);
     }
 
-    private boolean isFirstMove(GameBean game) {
-        return game.getCurrentMove() == 1;
+    private Integer validateIdIsInteger(String idString) throws PlayException {
+        try {
+            if (idString == null || idString.trim().length() == 0) {
```

Comment:
we can just do parseInt, if it null or empty, it should throw an exception, why do we need this logic?